src/App.jsx

```diff
@@ -1,23 +1,29 @@
-import { Routes, Route, } from "react-router-dom";
-import React, { useEffect, useState } from 'react'
-import './App.css'
-import Header from "./components/Header"
-import ListArticles from './components/ListArticles'
-import ArticleCard from "./components/ArticleCard";
+import "./App.css";
+import Header from "./components/Header";
+import Footer from "./components/Footer";
 import Nav from "./components/Nav";
+import ListArticles from "./components/ListArticles";
+import UserLogin from "./components/UserLogin";
+import "bootstrap/dist/css/bootstrap.min.css";
+import { Routes, Route } from "react-router-dom";
+import SingleArticleCard from "./components/SingleArticleCard";
+import ListComments from "./components/ListComments";
 
-function App() {
 
+function App() {
   return (
-    <div className="App">
-      <Header></Header>
-      <Nav></Nav>
+    <div className="app">
+      <UserLogin />
+      <Header />
+      <Nav />
       <Routes>
-        <Route path="/" element={<ListArticles />} />
+        <Route path="/" element={<ListArticles />}></Route>
+        <Route path="/article/:article_id" element={<SingleArticleCard />}></Route>
+        <Route path="/article/:article_id" element={<ListComments />}></Route>
```

Comment:
Instead of trying this way - you have two paths matching - just render ListComments inside your SingleArticleCard component and then remove this line. It will do the job 👍